test/IntegrationTests/Helpers/TestHttpListener.cs

```diff
@@ -126,6 +126,12 @@ private void HandleHttpRequests()
             {
                 // we don't care about any exception when listener is stopped
             }
+            catch (Exception ex)
+            {
+                // somethig unexpected happened
+                // log instead of crashing the thread
```

Comment:
Why do not crash application?

Reply:
test failures work only well on the thread which started the test

I was getting cryptic messages that the test runner was not able to connect to something

it is something I encountered when working on the previous PR